src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,143 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+public class Parser {
+    private final TaskList tasks;
+    private final Ui Ui;
+    private final Storage storage;
+    private final Scanner myScanner;
+
+    /**
+     * Creates a Parser
+     *
+     * @param tasks     the TaskList
+     * @param Ui        the ui object
+     * @param storage   the storage object
+     * @param myScanner the created Scanner object
+     */
+    Parser(TaskList tasks, Ui Ui, Storage storage, Scanner myScanner) {
+        this.tasks = tasks;
+        this.Ui = Ui;
+        this.storage = storage;
+        this.myScanner = myScanner;
+    }
+
+    /**
+     * Parses the input given by the user
+     *
+     * @param inValue the initials of the string given by user
+     */
+    public void parseInput(String inValue) {
+        Task item;
+        switch (inValue) {
+        case "bye":
+            Ui.goodbye();
+            storage.saveTasksToFile(tasks);
+            System.exit(0);
+            return;
+
+        case "list":
+            Ui.tasksInList(tasks);
+            break;
+
+        case "mark":
+            int number = myScanner.nextInt();
```

Comment:
The naming can be more consistent to improve clarity (number is used here but numero is used at the later part)